src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -152,21 +157,45 @@ private static BidRequest fillExplicitParameters(BidRequest bidRequest, RoutingC
      * This method extracts parameters from http request and overrides corresponding attributes in {@link BidRequest}.
      */
     private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest request) {
-        final Site updatedSite = overrideSitePage(bidRequest.getSite(), request);
+        final Site updatedSite = overrideSite(bidRequest.getSite(), request);
         final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request);
         final Long updatedTimeout = updateTimeout(request);
 
         return updateBidRequest(bidRequest, updatedSite, updatedImp, updatedTimeout);
     }
 
-    private static Site overrideSitePage(Site site, HttpServerRequest request) {
+    private static Site overrideSite(Site site, HttpServerRequest request) {
         final String canonicalUrl = canonicalUrl(request);
-        if (StringUtils.isBlank(canonicalUrl)) {
-            return site;
-        }
 
-        final Site.SiteBuilder siteBuilder = site == null ? Site.builder() : site.toBuilder();
-        return siteBuilder.page(canonicalUrl).build();
+        Site.SiteBuilder siteBuilder;
+        if (site == null) {
+            siteBuilder = Site.builder();
+            if (StringUtils.isNotBlank(canonicalUrl)) {
+                siteBuilder.page(canonicalUrl);
+            }
+            return siteBuilder
+                    .ext(Json.mapper.valueToTree(ExtSite.of(1)))
+                    .build();
+        } else {
+            final ObjectNode siteExt = site.getExt();
+            if (siteExt == null || siteExt.get("amp") == null || siteExt.get("amp").intValue() != 1) {
```

Comment:
Last condition `siteExt.get("amp").intValue() != 1` is redundant - it may be the case that `ext.amp` is not even integer. It's up to `RequestValidator` to verify that.